Add: Survive The Nights

QuintenQVD0 opened the pull request:
# Description

Closes #49 

## Checklist for all submissions

<!-- insert X into the brackets to mark it as done (i.e. [x]). You can click preview to make the links appear clickable. -->

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:


<!-- You can erase the new egg submission template if you're not adding a completely new egg -->

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

scootz commented:
I would suggest adding a variables for "useSteamRelays", "showInPublicLobby", "welcomeMessage", "recurringWelcomeMessage", "pvpDisabled".    With my setup, I had to disable useSteamRelays for anyone to connect to the server.

scootz commented:
Something I did just notice now, and changed the other day but forgot to mention here.  QUERY_PORT is set to false. shouldn't it be user_editable?

QuintenQVD0 commented:
> Something I did just notice now, and changed the other day but forgot to mention here. QUERY_PORT is not user editable. Shouldn't it be set to true?

no, a port should never be editable by a user

QuintenQVD0 commented:
You can add comments to the egg, that is not a problem

scootz commented on survive_the_nights/egg-survive-the-nights.json:
```diff
@@ -0,0 +1,123 @@
+{
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PANEL",
+    "meta": {
+        "version": "PTDL_v2",
```
```suggestion
        "version": "PLCN_v1",
```

gOOvER replied:
why? 2 different eggs

scootz replied:
sorry, was editing both one for ptero and the other for pelican.

scootz commented on survive_the_nights/egg-survive-the-nights.json:
```diff
@@ -0,0 +1,123 @@
+            "field_type": "text"
+        },
+        {
+            "sort": 2,
```
```suggestion

```

scootz commented on survive_the_nights/egg-pterodactyl-survive-the-nights.json:
```diff
@@ -0,0 +1,114 @@
+            "user_viewable": true,
+            "user_editable": false,
+            "rules": "required|numeric",
+            "field_type": "text"
```
```suggestion
            "field_type": "text"
        },
        {
            "name": "Use steam relays",
            "description": "Use the steam relays to bypass firewalls.",
            "env_variable": "USE_STEAM_RELAYS",
            "default_value": "True",
            "user_viewable": true,
            "user_editable": true,
            "rules": "required|string|in:True,False",
            "field_type": "text"
        },
        {
            "name": "Welcome message",
            "description": "Message shown to new players.",
            "env_variable": "WELCOME_MESSAGE",
            "default_value": "Welcome to the server.",
            "user_viewable": true,
            "user_editable": true,
            "rules": "string|nullable",
            "field_type": "text"
        },
        {
            "name": "Recurring welcome message",
            "description": "Message to display for players who have previously joined the server.",
            "env_variable": "RECURRING_WELCOME_MESSAGE",
            "default_value": "Welcome to the server.",
            "user_viewable": true,
            "user_editable": true,
            "rules": "string|nullable",
            "field_type": "text"
        },
        {
            "name": "PVP Disabled",
            "description": "Allow or disallow PVP",
            "env_variable": "PVP_DISABLED",
            "default_value": "False",
            "user_viewable": true,
            "user_editable": true,
            "rules": "required|string|in:True,False",
            "field_type": "text"
```

gOOvER replied:
???? Why that?

scootz commented on survive_the_nights/egg-survive-the-nights.json:
```diff
@@ -0,0 +1,123 @@
+            "field_type": "text"
+        },
+        {
+            "sort": 6,
```
```suggestion

```

QuintenQVD0 replied:
No. It is a pelican export.  So the sort stays.

scootz commented on survive_the_nights/egg-survive-the-nights.json:
```diff
@@ -0,0 +1,123 @@
+            "field_type": "text"
+        },
+        {
+            "sort": 7,
```
```suggestion

```

gOOvER replied:
why? pelican export this in this way

scootz commented on survive_the_nights/egg-pterodactyl-survive-the-nights.json:
```diff
@@ -0,0 +1,114 @@
+{
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PANEL",
```
```suggestion
    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
```

scootz commented on survive_the_nights/egg-survive-the-nights.json:
```diff
@@ -0,0 +1,123 @@
+            "field_type": "text"
+        },
+        {
+            "sort": 5,
```
```suggestion

```

scootz commented on survive_the_nights/egg-survive-the-nights.json:
```diff
@@ -0,0 +1,123 @@
+            "field_type": "text"
+        },
+        {
+            "sort": 4,
```
```suggestion

```

scootz commented on survive_the_nights/egg-survive-the-nights.json:
```diff
@@ -0,0 +1,123 @@
+        {
+            "sort": 7,
+            "name": "World name",
+            "description": "The name of the server world.\r\n\r\nDo not change this after the server started!",
```
```suggestion
            "description": "The name of the server world.\n\nDo not change this after the server started!",
```

scootz commented on survive_the_nights/egg-survive-the-nights.json:
```diff
@@ -0,0 +1,123 @@
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "nullable|string|max:32",
+            "field_type": "text"
```
```suggestion
            "rules": [
                "nullable",
                "string",
                "max:32"
            ],
            "sort": 4
```

scootz reviewed (changes requested):
These are the changes I think should be added/changed.

gOOvER commented:
> These are the changes I think should be added/changed.

You  know, that this is the Pelican egg? YOur requested changes are really senseless

scootz commented:
@gOOvER the PR has both eggs (ptero and pelican) I edited both to make it easier to approve the changes.  If this is wrong, i'm sorry :(  This is my first attempt at doing this ever. Again sorry, if it causes a problem.

scootz reviewed (changes requested):
Again sorry if this was wrong of me to do, this egg was in limbo since august last year, I saw that someone else requested an egg for this game and @QuintenQVD0 mentioned no one tested it. I tested it and works.. with the exception of useSteamRelays. I had to set that to false to get it to work.  I also added some variables.  If this is wrong again, i don't know how many times I've said it, but I am sorry.  purge my comments and forget about it.   I'm still working on a couple that say "pending" that won't clear for me. I'm new at this. I honestly don't know what I'm doing I guess.  

@gOOvER those suggested edits you ask why? I blame vscode.  I changed QUERY_PORT manually to false user_editable. Maybe VSCODE formatted it to remove spaces on save and thus the change. Noted. I'll refrain from allowing changes like that in the future. Sorry.

QuintenQVD0 commented:
> Again sorry if this was wrong of me to do, this egg was in limbo since august last year, I saw that someone else requested an egg for this game and @QuintenQVD0 mentioned no one tested it. I tested it and works.. with the exception of useSteamRelays. I had to set that to false to get it to work. I also added some variables. If this is wrong again, i don't know how many times I've said it, but I am sorry. purge my comments and forget about it. I'm still working on a couple that say "pending" that won't clear for me. I'm new at this. I honestly don't know what I'm doing I guess.
> 
> @gOOvER those suggested edits you ask why? I blame vscode. I changed QUERY_PORT manually to false user_editable. Maybe VSCODE formatted it to remove spaces on save and thus the change. Noted. I'll refrain from allowing changes like that in the future. Sorry.

Thank you for your revieuw. But you did it a bit in an unusual way. If you want extra variables leave a comment on the last line of the egg with just a list of them. Don't add them as full json in a comment. And same for the parser. Mention like the key in the file with var it should match. It should not have full code in a comment. 

Also just review the egg for the panel you have installed. Do not touch the other ones as the formats are different.

scootz commented:
> Thank you for your revieuw. But you did it a bit in an unusual way. If you want extra variables leave a comment on the last line of the egg with just a list of them. Don't add them as full json in a comment. And same for the parser. Mention like the key in the file with var it should match. It should not have full code in a comment.
> 
> Also just review the egg for the panel you have installed. Do not touch the other ones as the formats are different.

Mom always said I was unusual/different.  So what else is new? lol.  I do have both panels installed on two different computers, which is why I did both .json files.  The only time I ever directly edited the file was to change my original change for the query_port back to false for user_editable which is where some of the spacing changes may have happened when I saved the file.

gOOvER commented:
closed due to inactivity
